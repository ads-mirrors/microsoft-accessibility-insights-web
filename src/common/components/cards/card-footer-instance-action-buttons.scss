// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../styles/colors.scss';

.kebab-menu-icon {
    flex-grow: 1; // Center the custom SVG
    padding-left: 9px !important;
    padding-right: 9px !important;
    padding-top: 6px !important;
    padding-bottom: 6px !important;
    @media screen and (forced-colors: active) {
        color: buttontext !important;
    }
}

.fileissue {
    font-weight: 400 !important;

    &:hover {
        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }

    &:focus {
        outline: 1px solid;
    }
}

.copyfailuredetails {
    font-weight: 400 !important;

    &:hover {
        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }
}

.menu-Button {
    &:hover {
        border: 1px solid $primary-text !important;
        color: $primary-text !important;
    }

    &:focus {
        outline: 1px solid;
    }
}

.kebab-menu-button {
    margin-right: 8px;

    // Since the menu uses a custom SVG, set the button shape:
    width: 32px;
    height: 32px;

    &:hover {
        background-color: $menu-item-background-hover;
    }

    &:global(.is-expanded) {
        background-color: $menu-item-background-active;
    }

    // Add hover and active state:
    &:global(.is-expanded) .kebab-menu-icon,
    &:hover .kebab-menu-icon {
        @media screen and (forced-colors: active) {
            color: highlight;
        }
    }
}
