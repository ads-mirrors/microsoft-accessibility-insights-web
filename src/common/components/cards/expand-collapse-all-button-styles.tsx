// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { makeStyles, tokens } from '@fluentui/react-components';

export const useExpandCollapseAllButtonStyles: any = makeStyles({
    expandCollapseAllButton: {
        display: 'flex',
        padding: 'unset',
        paddingRight: '27px!important',
        paddingLeft: '1px!important',
        marginLeft: '1px!important',
        marginTop: '2px!important',
        fontWeight: tokens?.fontSizeBase400,
    },

    customStyleIcon: {
        color: tokens.colorNeutralForeground2BrandHover,
    },
});
