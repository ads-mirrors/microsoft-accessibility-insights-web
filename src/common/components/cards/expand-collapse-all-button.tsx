// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { useExpandCollapseAllButtonStyles } from 'common/components/cards/expand-collapse-all-button-styles';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';

export type ExpandCollapseAllButtonProps = {
    allCardsCollapsed: boolean;
    cardSelectionMessageCreator: CardSelectionMessageCreator;
};

export const ExpandCollapseAllButton = NamedFC<ExpandCollapseAllButtonProps>(
    'ExpandCollapseAllButton',
    props => {
        const getStyles: any = useExpandCollapseAllButtonStyles();
        const { allCardsCollapsed, cardSelectionMessageCreator } = props;

        let expandCollapseAllButtonHandler = cardSelectionMessageCreator.collapseAllRules;
        let buttonText = 'Collapse all';
        let ariaLabel: string | undefined = undefined;

        if (allCardsCollapsed) {
            expandCollapseAllButtonHandler = cardSelectionMessageCreator.expandAllRules;
            buttonText = 'Expand all';
            ariaLabel = 'Expand all rules to show failed instances.';
        }

        const IconName = () => {
            if (allCardsCollapsed) {
                return (
                    <FluentUIV9Icon
                        iconName="ChevronRight32Regular"
                        customClass={getStyles?.customStyleIcon}
                    />
                );
            }
            return (
                <FluentUIV9Icon
                    iconName="ChevronDown32Regular"
                    customClass={getStyles?.customStyleIcon}
                />
            );
        };

        return (
            <Button
                icon={<IconName />}
                size="medium"
                appearance="transparent"
                className={getStyles?.expandCollapseAllButton}
                aria-label={ariaLabel}
                aria-expanded={!allCardsCollapsed}
                onClick={expandCollapseAllButtonHandler}
            >
                {buttonText}
            </Button>
        );
    },
);
