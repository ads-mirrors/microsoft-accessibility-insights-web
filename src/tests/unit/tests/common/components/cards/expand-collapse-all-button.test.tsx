// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';

import { ExpandCollapseAllButton } from 'common/components/cards/expand-collapse-all-button';
import { AutomatedChecksCardSelectionMessageCreator } from 'common/message-creators/automated-checks-card-selection-message-creator';
import * as React from 'react';
import { It, IMock, Mock } from 'typemoq';
import {
    mockReactComponents,
    useOriginalReactElements,
} from '../../../../mock-helpers/mock-module-helpers';

jest.mock('@fluentui/react-components');
jest.mock('common/components/cards/expand-collapse-all-button-styles', () => {
    return {
        useExpandCollapseAllButtonStyles: jest.fn(),
    };
});
jest.mock('common/icons/fluentui-v9-icons');

describe('ExpandCollapseAllButton', () => {
    mockReactComponents([Button]);
    let cardSelectionMessageCreatorMock: IMock<AutomatedChecksCardSelectionMessageCreator>;

    beforeEach(() => {
        cardSelectionMessageCreatorMock = Mock.ofType(AutomatedChecksCardSelectionMessageCreator);
    });

    it.each([true, false])(
        'renders per snapshot with allCardsCollapsed %p',
        (allCardsCollapsed: boolean) => {
            const renderResult = render(
                <ExpandCollapseAllButton
                    allCardsCollapsed={allCardsCollapsed}
                    cardSelectionMessageCreator={cardSelectionMessageCreatorMock.object}
                />,
            );
            expect(renderResult.asFragment()).toMatchSnapshot();
        },
    );

    it('sends an expandAllRules message when clicked with all cards collapsed', async () => {
        cardSelectionMessageCreatorMock.setup(mock => mock.expandAllRules(It.isAny())).verifiable();
        useOriginalReactElements('@fluentui/react-components', ['Button']);
        const renderResult = render(
            <ExpandCollapseAllButton
                cardSelectionMessageCreator={cardSelectionMessageCreatorMock.object}
                allCardsCollapsed={true}
            />,
        );

        await userEvent.click(renderResult.getByRole('button'));

        cardSelectionMessageCreatorMock.verifyAll();
    });

    it('sends a collapseAllRules message when clicked with some cards expanded', async () => {
        cardSelectionMessageCreatorMock
            .setup(mock => mock.collapseAllRules(It.isAny()))
            .verifiable();
        useOriginalReactElements('@fluentui/react-components', ['Button']);
        const renderResult = render(
            <ExpandCollapseAllButton
                cardSelectionMessageCreator={cardSelectionMessageCreatorMock.object}
                allCardsCollapsed={false}
            />,
        );

        await userEvent.click(renderResult.getByRole('button'));

        cardSelectionMessageCreatorMock.verifyAll();
    });
});
