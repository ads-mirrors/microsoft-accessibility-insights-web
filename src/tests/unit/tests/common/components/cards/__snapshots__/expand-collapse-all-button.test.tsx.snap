// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandCollapseAllButton renders per snapshot with allCardsCollapsed false 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-expanded="true"
    icon="[object Object]"
    size="medium"
  >
    Collapse all
  </mock-button>
</DocumentFragment>
`;

exports[`ExpandCollapseAllButton renders per snapshot with allCardsCollapsed true 1`] = `
<DocumentFragment>
  <mock-button
    appearance="transparent"
    aria-expanded="false"
    aria-label="Expand all rules to show failed instances."
    icon="[object Object]"
    size="medium"
  >
    Expand all
  </mock-button>
</DocumentFragment>
`;
